src/Lib.hs

```diff
@@ -186,24 +189,16 @@ mkApp configuration = do
     convertToAppUsers (AdminUser username password) = ApplicationUser username password
 
 --runPoolInsertion poolMetadataJsonPath poolHash
-runPoolInsertion :: FilePath -> PoolId -> PoolMetadataHash -> IO (Either DBFail Text)
-runPoolInsertion poolMetadataJsonPath poolId poolHash = do
-    putTextLn $ "Inserting pool! " <> (toS poolMetadataJsonPath) <> " " <> (show poolId)
-
-    let dataLayer :: DataLayer
-        dataLayer = postgresqlDataLayer
+runPoolInsertion :: DataLayer -> Text -> PoolId -> PoolMetadataHash -> IO (Either DBFail Text)
+runPoolInsertion dataLayer poolMetadataJson poolId poolHash = do
+    putTextLn $ "Inserting pool! " <> (show poolId)
 
-    --PoolHash -> ByteString -> IO (Either DBFail PoolHash)
-    poolMetadataJson <- readFile poolMetadataJsonPath
-
-    -- Let us try to decode the contents to JSON.
     decodedMetadata <-  case (eitherDecode' $ BL.fromStrict (encodeUtf8 poolMetadataJson)) of
                             Left err     -> panic $ toS err
                             Right result -> return result
 
-    let addPoolMetadata = dlAddPoolMetadata dataLayer
-
-    addPoolMetadata Nothing poolId poolHash poolMetadataJson (pomTicker decodedMetadata)
+    dlAddPoolMetadata dataLayer Nothing poolId poolHash poolMetadataJson
```

Comment:
Please expand this to what it was before. I really don't like inlining too much since it hurts readability and we spend most of our time reading so...

Reply:
I don't understand what you mean with **inlining**?

Reply:
Trying to add too much into a single line.
The let binding is helpful as it makes it more clear what is it that we are doing.